pytest_html/plugin.py

```diff
@@ -490,9 +490,12 @@ def generate_summary_item(self):
             __name__, os.path.join("resources", "main.js")
         ).decode("utf-8")
 
+        report_title = os.path.basename(self.logfile)
+        session.config.hook.pytest_html_results_report_title(report_title=report_title)
```

Comment:
Isn't the behavior for this a bit confusing? I thought strings were immutable.

```
>>> def swap(rt):
...     rt = 'hi'
...
>>> rt = 'report title'
>>> swap(rt)
>>> rt
'report title'
>>>            
```

How can the hook modify the report_title in place? Shouldn't the hook return a value instead?

Reply:
Indeed, did think it about enough, sorry.

I see two solutions:
- the hook return a value but it doesn't work as the other hooks.
- the hook takes a mutable as input and we keep this behavior

I think the second option is better, and I'll push a commit to give the whole html title to the hook. It will allow more title customization.
Just need to document it, add test and examples before submitting again.